frontend/ui/widget/inputdialog.lua

```diff
@@ -528,10 +553,28 @@ function InputDialog:onShowKeyboard(ignore_first_hold_release)
     end
 end
 
+function InputDialog:toggleKeyboard(force_hide)
+    if force_hide and self.keyboard_hidden then return end
+    self.keyboard_hidden = not self.keyboard_hidden
+    self.input = self:getInputText() -- re-init with up-to-date text
+    self:onClose() -- will close keyboard and save view position
+    self:free()
```

Comment:
Haven't tested how addWidgets() added widget work in full screen (allowing toggling keyboard) mode, have you ? :)
Generically, shouldn't you `-- prevent these externally added widgets from being freed as :init() will re-add them` here too (even if we currently don't add any) ?

Reply:
> Haven't tested how addWidgets() added widget work in full screen (allowing toggling keyboard) mode, have you ?

Text editor? Find and Go dialogs work fine.

Reply:
Or you mean adding external widget to the Text editor frame itself? I haven't tried.

Reply:
I meant in the main TextEditor (with the navbar, etc...), addWidget stuff above that nav bar.
But I guess as we don't know the added widget when we compute all the heights, it would not work - so forget it :)

Reply:
<kbd>![2](https://user-images.githubusercontent.com/62179190/131823953-5985915f-4d42-472e-af07-2cb9f51791da.png)</kbd>

Wrong height (cut a half of checkbutton height at top and bottom) and doesn't survive keyboard toggling.